File: src/dev/storage/CXLDRAMInterface.py
from m5.objects.CXLMemCtrl import CXLMemCtrl
from m5.objects.DRAMInterface import *

class CXLDRAMInterface(DRAMInterface):
    type = "CXLDRAMInterface"
    cxx_header = "dev/storage/cxl_dram_interface.hh"
    cxx_class = "gem5::CXLDRAMInterface"
    
    def controller(self):
        controller = CXLMemCtrl()
        controller.dram = self
        return controller

File: src/dev/storage/cxl_dram_interface.cc
#include "dev/storage/cxl_dram_interface.hh"
#include "mem/dram_interface.hh"

namespace gem5
{

CXLDRAMInterface::CXLDRAMInterface(const CXLDRAMInterfaceParams &p):
    memory::DRAMInterface::DRAMInterface(p)
{

}
}